pkg/trace/agent/agent.go

```diff
@@ -242,11 +245,13 @@ func (a *Agent) Process(p *api.Payload) {
 
 		// Extra sanitization steps of the trace.
 		for _, span := range chunk.Spans {
-			for k, v := range a.conf.GlobalTags {
-				if k == tagOrigin {
-					chunk.Origin = v
-				} else {
-					traceutil.SetMeta(span, k, v)
+			if span.Meta[tagInferredSpanTagSourceKey] != "self" {
```

Comment:
Please add a comment below this line explaining why these spans don't receive global tags.